cs4360/libs/crm/shared/src/lib/dto/contact.dto.ts

```diff
@@ -1,18 +1,34 @@
-import {PhoneDTO} from "./phone.dto";
-import {EmailDTO} from "./email.dto";
-
+import {IsEmail, IsNotEmpty} from "class-validator";
 
 export interface ContactDTO {
 
   id: string;
 
+  // Properties
   firstName: string;
   lastName: string;
+  company? : string
+  notes?: string;
 
+  // Relations
   emails?: EmailDTO[];
   phones?: PhoneDTO[];
-  company? : string
+}
 
-  notes?: string;
+export class EmailDTO {
+  id: string;
+
+  @IsEmail()
+  address: string;
+
+  type?: string;
+  contact: ContactDTO;
+}
+
+export interface PhoneDTO {
+  id: string;
+  number: number;
+  type?: string;
+  contact: ContactDTO;
```

Comment:
I moved these here to try and get rid of a circular dependency, but I'm not convinced its a good idea.

@jjohnson509 @vinsonchin @GitNickProgramming @joliver2 Thoughts?

Reply:
To be clear, it did not fix the issue

Reply:
I think the `forwardRef()` from nestjs methods might help here. I tried to implement them at some point, but could never find the right place for them. My thought was that we should be able to forward the reference of the entities and that should be able to get rid of the issue.